core/src/main/java/bio/terra/pearl/core/service/kit/KitRequestService.java

```diff
@@ -272,18 +278,40 @@ protected String stringifyPepperAddress(PepperKitAddress kitAddress) {
      * to allow the rest of the batch to be processed.
      */
     private void saveKitStatus(KitRequest kit, PepperKit pepperKit, Instant pepperStatusFetchedAt) {
+        KitRequestStatus priorStatus = kit.getStatus();
         try {
             kit.setExternalKit(objectMapper.writeValueAsString(pepperKit));
             kit.setExternalKitFetchedAt(pepperStatusFetchedAt);
             kit.setStatus(PepperKitStatus.mapToKitRequestStatus(pepperKit.getCurrentStatus()));
             dao.update(kit);
         } catch (JsonProcessingException e) {
-            logger.warn(
-                    "Unable to serialize status JSON for kit %s: %s".formatted(kit.getId(), pepperKit.toString()),
-                    e);
+            log.error("Unable to serialize status JSON for kit %s: %s".formatted(kit.getId(), pepperKit.toString()), e);
+        }
+        try {
+            notifyKitStatus(kit, priorStatus);
+        } catch (Exception e) {
+            log.error("Error publishing kit status event for KitRequest enrollee %s".formatted(kit.getEnrolleeId()), e);
         }
     }
 
+    protected void notifyKitStatus(KitRequest kitRequest, KitRequestStatus priorStatus) {
+        // only notify on status change
+        if (priorStatus == kitRequest.getStatus()) {
+            return;
+        }
+        if (!KitRequestStatus.SENT.equals(kitRequest.getStatus())) {
+            return;
+        }
+
+        Enrollee enrollee = enrolleeService.find(kitRequest.getEnrolleeId()).orElseThrow(() ->
+                new IllegalStateException("Invalid enrollee for KitRequest %s: enrollee ID=%s".formatted(kitRequest.getId(), kitRequest.getEnrolleeId())));
+        List<PortalParticipantUser> ppUsers = portalParticipantUserService.findByParticipantUserId(enrollee.getParticipantUserId());
+        if (ppUsers.size() != 1) {
```

Comment:
this isn't an illegal state -- this is what we would expect if the same person (same email address) joined two different portals.  If this happens, we just need to identify which ppUser is the one that was used in creating this enrollee.  This can be done by find the ppUser with  `ppUser.profileId == enrollee.profileId`. Now if *that* returns multiple matches, we have an illegalState

Reply:
Incidentally, the ppUser<->enrollee relationship is one of the most confusing, because we didn't (and still don't) have a great sense of the arity of that relationship and how multi-study-multi-portal-mega-hub will impact it.  So I would be a big fan of taking the logic for navigating from an enrollee to a PortalParticipantUser and putting it in a `findPortalParticipantUser(Enrollee enrollee)` function in EnrolleeService so other folks can use it and it will be easier to update it as "hub" concepts clarify.

Reply:
Yup, thx.
I forgot to highlight this in the PR (too much time since I wrote it), since I was unsure and wanted feedback.